File: backend/src/bond_monitor/domain/notifications/policies.py
"""Notification delivery policies."""

from __future__ import annotations

from dataclasses import dataclass
from typing import Literal

AlertUrgency = Literal["normal", "soon", "critical"]


@dataclass(frozen=True)
class NotificationPolicy:
    put_offer_telegram_cooldown_hours: int = 24
    risk_telegram_min_urgency: AlertUrgency = "critical"
    include_put_offer_watch_in_alerts: bool = False


DEFAULT_NOTIFICATION_POLICY = NotificationPolicy()


def telegram_urgency_allowed(urgency: AlertUrgency) -> bool:
    return urgency == "critical"

File: backend/internal/domain/notifications/policies.go
package notifications

// NotificationPolicy controls alert delivery behavior.
type NotificationPolicy struct {
	PutOfferTelegramCooldownHours int
	RiskTelegramMinUrgency        AlertUrgency
	IncludePutOfferWatchInAlerts  bool
}

var DefaultNotificationPolicy = NotificationPolicy{
	PutOfferTelegramCooldownHours: 24,
	RiskTelegramMinUrgency:        AlertUrgencyCritical,
	IncludePutOfferWatchInAlerts:  false,
}

// TelegramUrgencyAllowed returns whether urgency qualifies for Telegram in legacy policy.
func TelegramUrgencyAllowed(urgency AlertUrgency) bool {
	return urgency == AlertUrgencyCritical
}
